pkg/metrics/metrics.go

```diff
@@ -124,6 +143,14 @@ func NewServerMetrics() *ServerMetrics {
 				},
 				[]string{scheduleLabel},
 			),
+			restoreTotal: prometheus.NewCounterVec(
+				prometheus.CounterOpts{
+					Namespace: metricNamespace,
+					Name:      restoreTotal,
+					Help:      "Number of existing restores",
+				},
+				[]string{scheduleLabel},
+			),
```

Comment:
I think we need a corresponding `restoreDeletionTotal` with the appropriate register function, as well. Otherwise, the number of restores will always go up and we have no accounting for ones deleted.

There are 3 ways a restore can be deleted - via `velero restore delete`, via `velero backup delete` deleting all restores associated with the backup, and via garbage control, which also deletes all of a backup's existing restores.

L324 (on this branch) in backup_deletion_controller.go is where restores are deleted in the case of a backup being deleted, either explicitly or via garbage collection.

Deleting a restore via `velero restore delete` doesn't happen on a controller, though - it happens [here](https://github.com/heptio/velero/blob/master/pkg/cmd/cli/restore/delete.go#L108), on the client side. I'm honestly not sure how we communicate that to the velero server running in the cluster. We'll have to do some more research on this, I think.